modules/compute/gke-node-pool/variables.tf

```diff
@@ -171,10 +171,36 @@ variable "spot" {
   default     = false
 }
 
+# tflint-ignore: terraform_unused_declarations
 variable "compact_placement" {
-  description = "Places node pool's nodes in a closer physical proximity in order to reduce network latency between nodes."
+  description = "DEPRECATED: Use `placement_policy`"
   type        = bool
-  default     = false
+  default     = null
+  validation {
+    condition     = var.compact_placement == null
+    error_message = "`compact_placement` is deprecated. Use `placement_policy` instead"
+  }
+}
+
+variable "placement_policy" {
+  description = <<-EOT
+  Group placement policy to use for the node pool's nodes. `COMPACT` is the only supported value for `type` currently. `name` is the name of the placement policy.
+  It is assumed that the specified policy exists. To create a placement policy refer to https://cloud.google.com/sdk/gcloud/reference/compute/resource-policies/create/group-placement.
+  Beware of the restrictions for placement policies https://cloud.google.com/compute/docs/instances/placement-policies-overview#restrictions-compact-policies
+  EOT
+
+  type = object({
+    type = string
+    name = optional(string)
+  })
+  default = {
+    type = null
+    name = null
+  }
+  validation {
+    condition     = var.placement_policy.type == null || try(contains(["COMPACT"], var.placement_policy.type), false)
```

Comment:
nit: Should the validation look like following, as placement_policy can be null as well ? 

```
var.placement_policy == null || var.placement_policy.type == null || try(contains(["COMPACT"], var.placement_policy.type), false)
```

Reply:
`placement_policy` cannot null. Its default is an object. 
```
  default = {
    type = null
    name = null
  }
```